From bba95b621d2ecabdad817bc91379de94908cfc2c Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Wed, 16 Jul 2014 20:42:39 -0400
Subject: [md] dm-mpath: eliminate pg_ready() wrapper

Message-id: <1405543364-20901-36-git-send-email-snitzer@redhat.com>
Patchwork-id: 85918
O-Subject: [RHEL7.1 PATCH 35/40] dm mpath: eliminate pg_ready() wrapper
Bugzilla: 1117872
RH-Acked-by: Mikulas Patocka <mpatocka@redhat.com>
RH-Acked-by: Heinz Mauelshagen <heinzm@redhat.com>

BZ: 1117872

Upstream linux-dm.git commit 0e799cb8a5ea4e2be592c28929f4ed5f498f1d19
Author: Mike Snitzer <snitzer@redhat.com>
Date:   Tue Jul 8 11:55:09 2014 -0400

    dm mpath: eliminate pg_ready() wrapper

    pg_ready() is not comprehensive in its logic and only serves to
    obfuscate code.  Replace pg_ready() with the appropriate logic in
    multipath_map().

    Signed-off-by: Mike Snitzer <snitzer@redhat.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/md/dm-mpath.c b/drivers/md/dm-mpath.c
index 3159ee1..de2f97f 100644
--- a/drivers/md/dm-mpath.c
+++ b/drivers/md/dm-mpath.c
@@ -373,8 +373,6 @@ static int __must_push_back(struct multipath *m)
    dm_noflush_suspending(m->ti)));
 }
 
-#define pg_ready(m) (!(m)->queue_io && !(m)->pg_init_required)
-
 /*
  * Map cloned requests
  */
@@ -402,11 +400,11 @@ static int multipath_map(struct dm_target *ti, struct request *clone,
   if (!__must_push_back(m))
    r = -EIO; /* Failed */
   goto out_unlock;
- }
- if (!pg_ready(m)) {
+ } else if (m->queue_io || m->pg_init_required) {
   __pg_init_all_paths(m);
   goto out_unlock;
  }
+
  if (set_mapinfo(m, map_context) < 0)
   /* ENOMEM, requeue */
   goto out_unlock;
-- 
1.7.1